### ISSUE TITLE
eslint/prefer-destructuring with re-assignment

### ISSUE THREAD

**Issue by mesqueeb** [oxc-project/oxc#12855]
if tells me to use destructuring with code like this:
```ts
  let endDate = stopDate

  // check to see if there is a date limit on the recurrences
  if (template.finishRecurOption === 'on' && template.endDate) {
    endDate = template.endDate
  }
```
at `endDate = template.endDate` specifically...
but that's not possible, right?

**Comment by huangtiandi1999** [comment]
Thanks for the report! 

1. Currently, oxlint's implementation of the`prefer-destructuring` rule is fully aligned with ESLint.
2. I understand the confusion — reporting on `AssignmentExpression` like `a = obj.a` or `a = arr[0]` can indeed be misleading, especially since such cases often can't be safely or meaningfully auto-fixed using destructuring.
3. You can configure oxlint (just like ESLint) to apply the rule only to `VariableDeclarator`, and skip `AssignmentExpression`. Here's a recommended configuration:
```json
{
    "prefer-destructuring": ["error", { "VariableDeclarator": { "object": true, "array": true }}],
}
```

**Comment by camc314** [comment]
thanks @huangtiandi1999 

Just to add,
In the example you've provided, it can be rewritten as:
```
  let endDate = stopDate

  // check to see if there is a date limit on the recurrences
  if (template.finishRecurOption === 'on' && template.endDate) {
    ({ endDate } = template);
  }
```

which should supress the error. Alternativly, as huangtiandi1999 mentions, provide options to the rule to allow this case.
